main.tf

```diff
@@ -244,40 +252,69 @@ resource "google_storage_bucket" "gitlab-runner-cache" {
   location      = var.region
   force_destroy = var.allow_force_destroy
 }
-// GKE Cluster
-module "gke" {
-  source  = "terraform-google-modules/kubernetes-engine/google"
-  version = "~> 12.0"
-
-  # Create an implicit dependency on service activation
-  project_id = module.project_services.project_id
 
-  name               = "gitlab"
-  region             = var.region
-  regional           = true
-  kubernetes_version = var.gke_version
+# GKE cluster
+resource "google_container_cluster" "gitlab" {
```

Comment:
Was there an issue with the upstream gke module?

Reply:
`google_project_service` is part of the official Google provider, and in my opinion has a couple of advantages:
* enables fine-grained dependencies
* makes maintenance easier by avoid to manage module versions

Reply:
I think this was the key change to be able to remove the `sleep_for_cluster_fix_helm_6361` hack.

Reply:
> manages the cluster and the node pool independently as recommended

The module does manages cluster and node_pools independently via `node_pools`

I think the `sleep_for_cluster_fix_helm_6361` should also be fixed by upgrading to latest. I recently tested the latest helm provider on latest gke module. If we run into issues, happy to fix them upstream.